rush-plugins/rush-migrate-subspace-plugin/src/cleanSubspace.ts

```diff
@@ -1,14 +1,243 @@
-import { chooseSubspacePrompt } from './prompts/subspace';
+import {
+  chooseSubspacePrompt,
+  scanForUnusedDependencyVersionsPrompt,
+  scanForDuplicatedDependenciesPrompt,
+  scanForSupersetDependencyVersionsPrompt,
+  scanForAllDependenciesPrompt
+} from './prompts/subspace';
 import Console from './providers/console';
 import { getRootPath } from './utilities/path';
 import { Colorize } from '@rushstack/terminal';
 import {
-  cleanSubspaceCommonVersions,
   getRushSubspaceCommonVersionsFilePath,
-  isSubspaceSupported
+  getSubspaceDependencies,
+  isSubspaceSupported,
+  loadRushSubspaceCommonVersions,
+  queryProjectsFromSubspace
 } from './utilities/subspace';
 import { getRushSubspacesConfigurationJsonPath, querySubspaces } from './utilities/repository';
 import { RushConstants } from '@rushstack/rush-sdk';
+import { chooseDependencyPrompt, confirmNextDependencyPrompt } from './prompts/dependency';
+import { IPackageJson, JsonFile } from '@rushstack/node-core-library';
+import { rSortVersions, subsetVersion } from './utilities/dependency';
```

Comment:
The name `rSortVersions` isn't following our coding standards because `r` is not an English word.  I had to go dig up that other file to realize that it means "reverse"